specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2021-05-13-preview/desktopvirtualization.json

```diff
@@ -3637,181 +3966,496 @@
+            "Cancel",
+            "Retry",
+            "Resume"
+          ],
+          "x-ms-enum": {
+            "name": "HostPoolUpdateAction",
             "modelAsString": true,
             "values": [
               {
-                "value": "Standard_LRS",
-                "description": "Standard HDD locally redundant storage. Best for backup, non-critical, and infrequent access."
+                "value": "Start",
+                "description": "Start the hostpool update."
               },
               {
-                "value": "Premium_LRS",
-                "description": "Premium SSD locally redundant storage. Best for production and performance sensitive workloads."
+                "value": "Pause",
+                "description": "Pause the hostpool update."
               },
               {
-                "value": "StandardSSD_LRS",
-                "description": "Standard SSD locally redundant storage. Best for web servers, lightly used enterprise applications and dev/test."
+                "value": "Cancel",
+                "description": "Cancel the hostpool update."
               },
               {
-                "value": "UltraSSD_LRS",
-                "description": "Ultra SSD locally redundant storage. Best for IO-intensive workloads such as SAP HANA, top tier databases (for example, SQL, Oracle), and other transaction-heavy workloads."
+                "value": "Retry",
+                "description": "Retry the hostpool update."
+              },
+              {
+                "value": "Resume",
+                "description": "Resume the hostpool update."
               }
             ]
           },
-          "description": "The disk type used by virtual machine in hostpool session host."
-        },
-        "vmCustomConfigurationUri": {
           "type": "string",
-          "description": "The uri to the storage blob containing scripts to be run on the virtual machine after provisioning."
-        },
-        "imageInfo": {
-          "$ref": "#/definitions/ImageInfoProperties",
-          "description": "Image configurations of HostPool."
-        },
-        "domainInfo": {
-          "$ref": "#/definitions/DomainInfoProperties",
-          "description": "Domain configurations of HostPool."
+          "description": "Action types for controlling hostpool update.",
+          "x-nullable": false
         }
       }
     },
-    "ImageInfoProperties": {
-      "description": "Image configurations of session host in a HostPool.",
+    "HostPoolUpdateProgress": {
+      "description": "Properties of host pool update progress.",
       "type": "object",
+      "readOnly": true,
       "properties": {
-        "type": {
+        "updateStatus": {
           "enum": [
-            "Gallery",
-            "StorageBlob",
-            "CustomImage"
+            "Scheduled",
+            "InProgress",
+            "Pausing",
+            "Paused",
+            "StartFailed",
```

Comment:
When/what situation to use "StartFailed"?

Reply:
@shailahirji